SimTracker/SiPixelDigitizer/plugins/SiPixelChargeReweightingAlgorithm.cc

```diff
@@ -562,3 +572,203 @@ void SiPixelChargeReweightingAlgorithm::printCluster(float arr[TXSIZE][TYSIZE])
+  std::vector<PixelDigi>::const_iterator loopDigi;
+  for (loopDigi = digis.begin(); loopDigi != digis.end(); ++loopDigi) {
+    unsigned int chan = loopDigi->channel();
+    if (loopTempSH.isInTheList(chan)) {
+      float corresponding_charge = loopDigi->adc();
+      theDigiSignal[chan] += SiPixelDigitizerAlgorithm::Amplitude(corresponding_charge, corresponding_charge);
+      chargeBefore += corresponding_charge;
+      if (loopDigi->row() < irow_min)
+        irow_min = loopDigi->row();
+      if (loopDigi->row() > irow_max)
+        irow_max = loopDigi->row();
+      if (loopDigi->column() < icol_min)
+        icol_min = loopDigi->column();
+      if (loopDigi->column() > icol_max)
+        icol_max = loopDigi->column();
+    }
+  }
+  // end loop on digis
+
+  LocalVector direction = loopTempSH.exitPoint() - loopTempSH.entryPoint();
+  LocalPoint hitEntryPoint = loopTempSH.entryPoint();
+  float trajectoryScaleToPosition = hitEntryPoint.z() / direction.z();
+  if ((hitEntryPoint.z() > 0 && direction.z() < 0) || (hitEntryPoint.z() < 0 && direction.z() > 0)) {
+    trajectoryScaleToPosition *= -1;
+  }
+  LocalPoint hitPosition = hitEntryPoint + trajectoryScaleToPosition * direction;
+
+  // addition as now the hit himself is not available
+  // https://github.com/cms-sw/cmssw/blob/master/SimDataFormats/TrackingHit/interface/PSimHit.h#L52
+  LocalPoint hitLocalPosition = hitEntryPoint + 0.5 * direction;
+  MeasurementPoint hitPositionPixel = topol->measurementPosition(hitLocalPosition);
+  std::pair<int, int> hitPixel =
+      std::pair<int, int>(int(floor(hitPositionPixel.x())), int(floor(hitPositionPixel.y())));
+
+  MeasurementPoint originPixel = MeasurementPoint(hitPixel.first - THX + 0.5, hitPixel.second - THY + 0.5);
+  LocalPoint origin = topol->localPosition(originPixel);
+
+  MeasurementPoint hitEntryPointPixel = topol->measurementPosition(loopTempSH.entryPoint());
+  MeasurementPoint hitExitPointPixel = topol->measurementPosition(loopTempSH.exitPoint());
+  std::pair<int, int> entryPixel =
+      std::pair<int, int>(int(floor(hitEntryPointPixel.x())), int(floor(hitEntryPointPixel.y())));
+  std::pair<int, int> exitPixel =
+      std::pair<int, int>(int(floor(hitExitPointPixel.x())), int(floor(hitExitPointPixel.y())));
+
+  int hitcol_min, hitcol_max, hitrow_min, hitrow_max;
+  if (entryPixel.first > exitPixel.first) {
+    hitrow_min = exitPixel.first;
+    hitrow_max = entryPixel.first;
+  } else {
+    hitrow_min = entryPixel.first;
+    hitrow_max = exitPixel.first;
+  }
+
+  if (entryPixel.second > exitPixel.second) {
+    hitcol_min = exitPixel.second;
+    hitcol_max = entryPixel.second;
+  } else {
+    hitcol_min = entryPixel.second;
+    hitcol_max = exitPixel.second;
+  }
+
+  if (!(irow_min <= hitrow_max && irow_max >= hitrow_min && icol_min <= hitcol_max && icol_max >= hitcol_min)) {
+    // The clusters do not have an overlap, hence the hit is NOT reweighted
+    return false;
+  }
+
+  float cmToMicrons = 10000.f;
```

Comment:
could be `static contexpr`. By the way wondering if there's already something similar defined on the rest of the digitization package.

Reply:
I can try with `static contexpr`.
 
I have just copied what was already used in [https://github.com/carolinecollard/cmssw/blob/additionalDigiInfo_12_4_X/SimTracker/SiPixelDigitizer/plugins/SiPixelChargeReweightingAlgorithm.cc#L214:L217]

Otherwise, there exists something similar in several places in the cmssw code : [https://github.com/cms-sw/cmssw/search?q=cmTomicron]

Reply:
@carolinecollard your links do not work.

Reply:
> I have just copied what was already used in [https://github.com/carolinecollard/cmssw/blob/additionalDigiInfo_12_4_X/SimTracker/SiPixelDigitizer/plugins/SiPixelChargeReweightingAlgorithm.cc#L214:L217]

right, ideally you want to put that in a single (central) place in the package a use it where necessary.

Reply:
ok. I will use it in a central place. 
For the link, I think I manage to do it correctly now :  [2nd link](https://github.com/cms-sw/cmssw/search?q=cmTomicron)